projects/manager/src/app/datasets/add-dataset/add-dataset.component.ts

```diff
@@ -0,0 +1,113 @@
+import {Component, ViewChild} from '@angular/core';
+import {FormControl, FormGroup, Validators} from '@angular/forms';
+import {DatasetsService, errorToText} from '@geoengine/common';
+import {DataPath} from '@geoengine/openapi-client';
+import {LoadingInfoComponent} from '../loading-info/loading-info.component';
+import {MatSnackBar} from '@angular/material/snack-bar';
+import {MatDialogRef} from '@angular/material/dialog';
+
+enum SourceOperators {
+    GdalSource,
+    OgrSource,
+}
+
+enum DataPaths {
+    Upload,
+    Volume,
+}
+
+export interface AddDatasetForm {
+    name: FormControl<string>;
+    displayName: FormControl<string>;
+    sourceOperator: FormControl<SourceOperators>;
+    dataPathType: FormControl<DataPaths>;
+    dataPath: FormControl<string>;
+}
+
+@Component({
+    selector: 'geoengine-manager-add-dataset',
+    templateUrl: './add-dataset.component.html',
+    styleUrl: './add-dataset.component.scss',
+})
+export class AddDatasetComponent {
+    DataPaths = DataPaths;
+    SourceOperators = SourceOperators;
+
+    @ViewChild(LoadingInfoComponent) loadingInfoComponent!: LoadingInfoComponent;
+
+    form: FormGroup<AddDatasetForm> = new FormGroup<AddDatasetForm>({
+        name: new FormControl('', {
+            nonNullable: true,
+            validators: [Validators.required, Validators.pattern(/^[a-zA-Z0-9_]+$/), Validators.minLength(3)],
+        }),
+        displayName: new FormControl('', {
+            nonNullable: true,
+            validators: [Validators.required],
+        }),
+        sourceOperator: new FormControl(SourceOperators.GdalSource, {
+            nonNullable: true,
+            validators: [Validators.required],
+        }),
+        dataPathType: new FormControl(DataPaths.Upload, {
+            nonNullable: true,
+            validators: [Validators.required],
+        }),
+        dataPath: new FormControl('', {
+            nonNullable: true,
+            validators: [Validators.required],
+        }),
+    });
+
+    constructor(
+        private readonly datasetsService: DatasetsService,
+        private readonly snackBar: MatSnackBar,
+        private dialogRef: MatDialogRef<AddDatasetComponent>,
+    ) {}
+
+    updateSourceOperator(): void {}
+
+    updateDataPath(): void {}
```

Comment:
What is this?

Reply:
e2266f3765f4e59bb5885efdebf24981df079969